src/core/ir/mod.rs

```diff
@@ -54,6 +56,41 @@ impl IR {
     }
 }
 
+#[derive(Clone, Debug)]
+pub struct Map {
+    pub input: Box<IR>,
+    // accept key return value instead of
+    pub map: HashMap<String, ConstValue>,
+}
+
+impl Eval for Map {
+    fn eval<'a, Ctx: ResolverContextLike<'a> + Sync + Send>(
+        &'a self,
+        ctx: EvaluationContext<'a, Ctx>,
+    ) -> Pin<Box<dyn Future<Output = Result<async_graphql::Value, EvaluationError>> + 'a + Send>>
+    where
+        async_graphql::Value: 'a,
+    {
+        Box::pin(async move {
+            let value = self.input.eval(ctx).await?;
+            if let ConstValue::String(key) = value {
+                if let Some(value) = self.map.get(&key) {
+                    Ok(value.to_owned())
+                } else {
+                    Err(EvaluationError::ExprEvalError(format!(
+                        "Can't find mapped key: {}.",
+                        key
+                    )))
+                }
+            } else {
+                Err(EvaluationError::ExprEvalError(
+                    "Mapped key must be string value.".to_owned(),
+                ))
```

Comment:
Based on my previous comment, we should be able to match with any arbitary ConstValue.

Reply:
Actually the implementation isn't right. 

We need to store a `HashMap<ConstValue, String>` so that the incoming value can be mapped to any EnumVariant.

Reply:
You are right, but we want to be able to map any ConstValue to a String. For I can do the following 

```graphql
enum Chars {
  A @alias(options: ["alpha", 1, {this_can_also_work: true}]),
  B,
  C
}

```

Reply:
Can use `serde_json::Value` also.

Reply:
In that case, lets completely drop const_value from the Map, keep it as a `HashMap<String, String>`